# 01. Data Overview

## Цель

Познакомиться с датасетом Olist, изучить структуру таблиц, определить связи между ними и выявить потенциальные проблемы данных.

In [2]:
import pandas as pd
import numpy as np

In [3]:
orders = pd.read_csv("../data/raw/olist_orders_dataset.csv")
customers = pd.read_csv("../data/raw/olist_customers_dataset.csv")
geolocation = pd.read_csv("../data/raw/olist_geolocation_dataset.csv")
order_items = pd.read_csv("../data/raw/olist_order_items_dataset.csv")
order_payments = pd.read_csv("../data/raw/olist_order_payments_dataset.csv")
order_reviews = pd.read_csv("../data/raw/olist_order_reviews_dataset.csv")
products = pd.read_csv("../data/raw/olist_products_dataset.csv")
sellers = pd.read_csv("../data/raw/olist_sellers_dataset.csv")
product_category_name_translation = pd.read_csv("../data/raw/product_category_name_translation.csv")

In [4]:
orders.head()

,order_id,customer_id,order_status,order_purchase_timestamp,order_approved_at,order_delivered_carrier_date,order_delivered_customer_date,order_estimated_delivery_date
0,e481f51cbdc54678b7cc49136f2d6af7,9ef432eb6251297304e76186b10a928d,delivered,2017-10-02 10:56:33,2017-10-02 11:07:15,2017-10-04 19:55:00,2017-10-10 21:25:13,2017-10-18 00:00:00
1,53cdb2fc8bc7dce0b6741e2150273451,b0830fb4747a6c6d20dea0b8c802d7ef,delivered,2018-07-24 20:41:37,2018-07-26 03:24:27,2018-07-26 14:31:00,2018-08-07 15:27:45,2018-08-13 00:00:00
2,47770eb9100c2d0c44946d9cf07ec65d,41ce2a54c0b03bf3443c3d931a367089,delivered,2018-08-08 08:38:49,2018-08-08 08:55:23,2018-08-08 13:50:00,2018-08-17 18:06:29,2018-09-04 00:00:00
3,949d5b44dbf5de918fe9c16f97b45f8a,f88197465ea7920adcdbec7375364d82,delivered,2017-11-18 19:28:06,2017-11-18 19:45:59,2017-11-22 13:39:59,2017-12-02 00:28:42,2017-12-15 00:00:00
4,ad21c59c0840e6cb83a9ceb5573f8159,8ab97904e6daea8866dbdbc4fb7aad2c,delivered,2018-02-13 21:18:39,2018-02-13 22:20:29,2018-02-14 19:46:34,2018-02-16 18:17:02,2018-02-26 00:00:00


In [5]:
customers.head()

,customer_id,customer_unique_id,customer_zip_code_prefix,customer_city,customer_state
0,06b8999e2fba1a1fbc88172c00ba8bc7,861eff4711a542e4b93843c6dd7febb0,14409,franca,SP
1,18955e83d337fd6b2def6b18a428ac77,290c77bc529b7ac935b93aa66c333dc3,9790,sao bernardo do campo,SP
2,4e7b3e00288586ebd08712fdd0374a03,060e732b5b29e8181a18229c7b0b2b5e,1151,sao paulo,SP
3,b2b6027bc5c5109e529d4dc6358b12c3,259dac757896d24d7702b9acbbff3f3c,8775,mogi das cruzes,SP
4,4f2d8ab171c80ec8364f7c12e35b23ad,345ecd01c38d18a9036ed96c73b8d066,13056,campinas,SP


In [6]:
geolocation.head()

,geolocation_zip_code_prefix,geolocation_lat,geolocation_lng,geolocation_city,geolocation_state
0,1037,-23.545621,-46.639292,sao paulo,SP
1,1046,-23.546081,-46.644820,sao paulo,SP
2,1046,-23.546129,-46.642951,sao paulo,SP
3,1041,-23.544392,-46.639499,sao paulo,SP
4,1035,-23.541578,-46.641607,sao paulo,SP


In [7]:
order_items.head()

,order_id,order_item_id,product_id,seller_id,shipping_limit_date,price,freight_value
0,00010242fe8c5a6d1ba2dd792cb16214,1,4244733e06e7ecb4970a6e2683c13e61,48436dade18ac8b2bce089ec2a041202,2017-09-19 09:45:35,58.90,13.29
1,00018f77f2f0320c557190d7a144bdd3,1,e5f2d52b802189ee658865ca93d83a8f,dd7ddc04e1b6c2c614352b383efe2d36,2017-05-03 11:05:13,239.90,19.93
2,000229ec398224ef6ca0657da4fc703e,1,c777355d18b72b67abbeef9df44fd0fd,5b51032eddd242adc84c38acab88f23d,2018-01-18 14:48:30,199.00,17.87
3,00024acbcdf0a6daa1e931b038114c75,1,7634da152a4610f1595efa32f14722fc,9d7a1d34a5052409006425275ba1c2b4,2018-08-15 10:10:18,12.99,12.79
4,00042b26cf59d7ce69dfabb4e55b4fd9,1,ac6c3623068f30de03045865e4e10089,df560393f3a51e74553ab94004ba5c87,2017-02-13 13:57:51,199.90,18.14


In [8]:
order_payments.head()

,order_id,payment_sequential,payment_type,payment_installments,payment_value
0,b81ef226f3fe1789b1e8b2acac839d17,1,credit_card,8,99.33
1,a9810da82917af2d9aefd1278f1dcfa0,1,credit_card,1,24.39
2,25e8ea4e93396b6fa0d3dd708e76c1bd,1,credit_card,1,65.71
3,ba78997921bbcdc1373bb41e913ab953,1,credit_card,8,107.78
4,42fdf880ba16b47b59251dd489d4441a,1,credit_card,2,128.45


In [9]:
order_reviews.head()

,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
0,7bc2406110b926393aa56f80a40eba40,73fc7af87114b39712e6da79b0a377eb,4,NaN,NaN,2018-01-18 00:00:00,2018-01-18 21:46:59
1,80e641a11e56f04c1ad469d5645fdfde,a548910a1c6147796b98fdf73dbeba33,5,NaN,NaN,2018-03-10 00:00:00,2018-03-11 03:05:13
2,228ce5500dc1d8e020d8d1322874b6f0,f9e4b658b201a9f2ecdecbb34bed034b,5,NaN,NaN,2018-02-17 00:00:00,2018-02-18 14:36:24
3,e64fb393e7b32834bb789ff8bb30750e,658677c97b385a9be170737859d3511b,5,NaN,Recebi bem antes do prazo estipulado.,2017-04-21 00:00:00,2017-04-21 22:02:06
4,f7c4243c7fe1938f181bec41a392bdeb,8e6bfb81e283fa7e4f11123a3fb894f1,5,NaN,Parabéns lojas lannister adorei comprar pela I...,2018-03-01 00:00:00,2018-03-02 10:26:53


In [10]:
products.head()

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
0,1e9e8ef04dbcff4541ed26657ea517e5,perfumaria,40.0,287.0,1.0,225.0,16.0,10.0,14.0
1,3aa071139cb16b67ca9e5dea641aaa2f,artes,44.0,276.0,1.0,1000.0,30.0,18.0,20.0
2,96bd76ec8810374ed1b65e291975717f,esporte_lazer,46.0,250.0,1.0,154.0,18.0,9.0,15.0
3,cef67bcfe19066a932b7673e239eb23d,bebes,27.0,261.0,1.0,371.0,26.0,4.0,26.0
4,9dc1a7de274444849c219cff195d0b71,utilidades_domesticas,37.0,402.0,4.0,625.0,20.0,17.0,13.0


In [11]:
sellers.head()

,seller_id,seller_zip_code_prefix,seller_city,seller_state
0,3442f8959a84dea7ee197c632cb2df15,13023,campinas,SP
1,d1b65fc7debc3361ea86b5f14c68d2e2,13844,mogi guacu,SP
2,ce3ad9de960102d0677a81f5d0bb7b2d,20031,rio de janeiro,RJ
3,c0f3eea2e14555b6faeea3dd58c1b1c3,4195,sao paulo,SP
4,51a04a8a6bdcb23deccc82b0b80742cf,12914,braganca paulista,SP


In [12]:
product_category_name_translation.head()

,product_category_name,product_category_name_english
0,beleza_saude,health_beauty
1,informatica_acessorios,computers_accessories
2,automotivo,auto
3,cama_mesa_banho,bed_bath_table
4,moveis_decoracao,furniture_decor


# Orders Dataset

In [13]:
orders.shape

(99441, 8)

In [14]:
orders.info()

<class 'pandas.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   order_id                       99441 non-null  str  
 1   customer_id                    99441 non-null  str  
 2   order_status                   99441 non-null  str  
 3   order_purchase_timestamp       99441 non-null  str  
 4   order_approved_at              99281 non-null  str  
 5   order_delivered_carrier_date   97658 non-null  str  
 6   order_delivered_customer_date  96476 non-null  str  
 7   order_estimated_delivery_date  99441 non-null  str  
dtypes: str(8)
memory usage: 6.1 MB


In [15]:
orders.isna().sum()

order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

In [16]:
orders.duplicated().sum()

np.int64(0)

In [17]:
orders["order_id"].is_unique

True

## Первичные наблюдения

- Таблица содержит информацию обо всех заказах интернет-магазина Olist.
- Каждая запись соответствует одному заказу, а столбец `order_id` является уникальным идентификатором заказа.
- Данные включают несколько временных меток, позволяющих проследить весь процесс выполнения заказа: от оформления до доставки клиенту.
- Полных дубликатов строк не обнаружено, что говорит о хорошем качестве данных.
- В столбцах, связанных с подтверждением и доставкой заказа, присутствуют пропуски. Предположительно они могут быть связаны с отменёнными заказами, заказами в процессе обработки или другими особенностями жизненного цикла заказа. Эта гипотеза будет проверена на следующих этапах анализа.
- Перед дальнейшим анализом потребуется преобразовать все столбцы с датами из строкового типа (`object`) в тип `datetime`.


# Customers Dataset

In [18]:
customers.shape

(99441, 5)

In [19]:
customers.info()

<class 'pandas.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype
---  ------                    --------------  -----
 0   customer_id               99441 non-null  str  
 1   customer_unique_id        99441 non-null  str  
 2   customer_zip_code_prefix  99441 non-null  int64
 3   customer_city             99441 non-null  str  
 4   customer_state            99441 non-null  str  
dtypes: int64(1), str(4)
memory usage: 3.8 MB


In [20]:
customers.isna().sum()

customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

In [21]:
customers.duplicated().sum()

np.int64(0)

In [22]:
customers["customer_id"].is_unique

True

## Первичные наблюдения

- Таблица содержит информацию о клиентах интернет-магазина Olist.
- Каждая строка соответствует одному экземпляру клиента в системе. Столбец customer_id является уникальным идентификатором записи.
- Помимо технического идентификатора customer_id, таблица содержит customer_unique_id, который позволяет идентифицировать одного и того же клиента между разными заказами.
- Полных дубликатов строк не обнаружено, что говорит о хорошем качестве данных.
- Ни в одном столбце пропусков не обнаружено.
- Типы данных соответствуют содержимому столбцов и не требуют преобразования на этапе очистки.


# Geolocation Dataset

In [23]:
geolocation.shape

(1000163, 5)

In [24]:
geolocation.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  str    
 4   geolocation_state            1000163 non-null  str    
dtypes: float64(2), int64(1), str(2)
memory usage: 38.2 MB


In [25]:
geolocation.isna().sum()

geolocation_zip_code_prefix    0
geolocation_lat                0
geolocation_lng                0
geolocation_city               0
geolocation_state              0
dtype: int64

In [26]:
geolocation.duplicated().sum()

np.int64(261831)

## Первичные наблюдения

- Таблица не содержит отдельного уникального идентификатора строки.
- Столбец `geolocation_zip_code_prefix` не является первичным ключом, поскольку один почтовый индекс может встречаться в нескольких записях.
- Каждая строка соответствует одной географической точке с координатами, городом и штатом.
- В таблице обнаружено 261 831 полностью дублирующаяся строка.
- Для таблицы `geolocation` это не обязательно свидетельствует об ошибке данных, так как одна и та же географическая точка может встречаться несколько раз.
- Решение об удалении дубликатов будет принято на этапе очистки данных после определения роли этой таблицы в анализе.


# Order_items Dataset

In [27]:
order_items.shape

(112650, 7)

In [28]:
order_items.info()

<class 'pandas.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  str    
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  str    
 3   seller_id            112650 non-null  str    
 4   shipping_limit_date  112650 non-null  str    
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), str(4)
memory usage: 6.0 MB


In [29]:
order_items.isna().sum()

order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

In [30]:
order_items.duplicated().sum()

np.int64(0)

### Проверка составного ключа

In [31]:
order_items[["order_id", "order_item_id"]].duplicated().sum()

np.int64(0)

## Первичные наблюдения

- Таблица содержит информацию о товарах, входящих в состав заказов.
- Каждая строка соответствует одному товару в конкретном заказе.
- Столбец `order_id` не является уникальным, поскольку один заказ может содержать несколько товаров.
- Столбец `order_item_id` также не является уникальным сам по себе — он представляет собой порядковый номер товара внутри заказа.
- Уникальной является комбинация столбцов `order_id` и `order_item_id`, которая выступает составным ключом таблицы.
- На этапе очистки данных столбец `shipping_limit_date` потребуется преобразовать из строкового типа `object` в тип `datetime`.
- Таблица содержит больше строк, чем таблица `orders`, что говорит о наличии заказов, включающих несколько товаров.

# Order_payments Dataset

In [32]:
order_payments.shape

(103886, 5)

In [33]:
order_payments.info()

<class 'pandas.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  str    
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  str    
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), str(2)
memory usage: 4.0 MB


In [34]:
order_payments.isna().sum()

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

In [35]:
order_payments.duplicated().sum()

np.int64(0)

### Проверка составного ключа

In [36]:
order_payments[["order_id", "payment_sequential"]].duplicated().sum()

np.int64(0)

## Первичные наблюдения

- Таблица содержит информацию обо всех платежах по заказам.
- Каждая строка соответствует одному платежу по заказу.
- Один заказ может быть оплачен несколькими платежами.
- Столбец `order_id` не является уникальным.
- Столбец `payment_sequential` представляет собой порядковый номер платежа внутри заказа и также не является уникальным сам по себе.
- Уникальной является комбинация столбцов `order_id` и `payment_sequential`, образующая составной ключ таблицы.
- Пропуски и полные дубликаты отсутствуют.

# Order_reviews Dataset

In [37]:
order_reviews.shape

(99224, 7)

In [38]:
order_reviews.info()

<class 'pandas.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   review_id                99224 non-null  str  
 1   order_id                 99224 non-null  str  
 2   review_score             99224 non-null  int64
 3   review_comment_title     11568 non-null  str  
 4   review_comment_message   40977 non-null  str  
 5   review_creation_date     99224 non-null  str  
 6   review_answer_timestamp  99224 non-null  str  
dtypes: int64(1), str(6)
memory usage: 5.3 MB


In [39]:
order_reviews.isna().sum()

review_id                      0
order_id                       0
review_score                   0
review_comment_title       87656
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

In [40]:
order_reviews.duplicated().sum()

np.int64(0)

In [41]:
order_reviews["review_id"].is_unique

False

In [42]:
order_reviews[["review_id", "order_id"]].duplicated().sum()

np.int64(0)

In [43]:
duplicates = order_reviews[
    order_reviews["review_id"].duplicated(keep=False)
]

duplicates.sort_values("review_id")

,review_id,order_id,review_score,review_comment_title,review_comment_message,review_creation_date,review_answer_timestamp
46678,00130cbe1f9d422698c812ed8ded1919,dfcdfc43867d1c1381bfaf62d6b9c195,1,NaN,"O cartucho ""original HP"" 60XL não é reconhecid...",2018-03-07 00:00:00,2018-03-20 18:08:23
29841,00130cbe1f9d422698c812ed8ded1919,04a28263e085d399c97ae49e0b477efa,1,NaN,"O cartucho ""original HP"" 60XL não é reconhecid...",2018-03-07 00:00:00,2018-03-20 18:08:23
90677,0115633a9c298b6a98bcbe4eee75345f,78a4201f58af3463bdab842eea4bc801,5,NaN,NaN,2017-09-21 00:00:00,2017-09-26 03:27:47
63193,0115633a9c298b6a98bcbe4eee75345f,0c9850b2c179c1ef60d2855e2751d1fa,5,NaN,NaN,2017-09-21 00:00:00,2017-09-26 03:27:47
92876,0174caf0ee5964646040cd94e15ac95e,f93a732712407c02dce5dd5088d0f47b,1,NaN,Produto entregue dentro de embalagem do fornec...,2018-03-07 00:00:00,2018-03-08 03:00:53
...,...,...,...,...,...,...,...
31120,fe5c833752953fed3209646f1f63b53c,4863e15fa53273cc7219c58f5ffda4fb,1,NaN,"Comprei dois produtos e ambos, mesmo enviados ...",2018-02-28 00:00:00,2018-02-28 13:57:52
7870,ff2fc9e68f8aabfbe18d710b83aabd30,2da58e0a7dcfa4ce1e00fad9d03ca3b5,2,NaN,NaN,2018-03-17 00:00:00,2018-03-19 11:44:15
82521,ff2fc9e68f8aabfbe18d710b83aabd30,1078d496cc6ab9a8e6f2be77abf5091b,2,NaN,NaN,2018-03-17 00:00:00,2018-03-19 11:44:15
73951,ffb8cff872a625632ac983eb1f88843c,c44883fc2529b4aa03ca90e7e09d95b6,3,NaN,NaN,2017-07-22 00:00:00,2017-07-26 13:41:07


In [44]:
order_reviews["review_score"].value_counts().sort_index()

review_score
1    11424
2     3151
3     8179
4    19142
5    57328
Name: count, dtype: int64

In [45]:
order_reviews["review_comment_message"].isna().mean()

np.float64(0.5870253164556962)

### Первичные наблюдения

- Таблица содержит информацию об отзывах покупателей.
- Каждая запись соответствует одному отзыву, связанному с конкретным заказом.
- Полных дубликатов не обнаружено.
- Столбец `review_id` не является уникальным идентификатором и встречается повторно.
- Комбинация `review_id` и `order_id` является уникальной.
- В столбцах `review_comment_title` и `review_comment_message` присутствует большое количество пропусков, что говорит о том, что многие пользователи оставляли только оценку без текстового комментария.
- Столбцы с датами (`review_creation_date`, `review_answer_timestamp`) имеют строковый тип и перед дальнейшим анализом должны быть преобразованы в `datetime`.

# Products Dataset

In [46]:
products.shape

(32951, 9)

In [47]:
products.info()

<class 'pandas.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  str    
 1   product_category_name       32341 non-null  str    
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), str(2)
memory usage: 2.3 MB


In [51]:
products["product_id"].is_unique

True

In [52]:
products.isna().sum()

product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

In [55]:
products[
    products["product_category_name"].isna()
]

,product_id,product_category_name,product_name_lenght,product_description_lenght,product_photos_qty,product_weight_g,product_length_cm,product_height_cm,product_width_cm
105,a41e356c76fab66334f36de622ecbd3a,NaN,NaN,NaN,NaN,650.0,17.0,14.0,12.0
128,d8dee61c2034d6d075997acef1870e9b,NaN,NaN,NaN,NaN,300.0,16.0,7.0,20.0
145,56139431d72cd51f19eb9f7dae4d1617,NaN,NaN,NaN,NaN,200.0,20.0,20.0,20.0
154,46b48281eb6d663ced748f324108c733,NaN,NaN,NaN,NaN,18500.0,41.0,30.0,41.0
197,5fb61f482620cb672f5e586bb132eae9,NaN,NaN,NaN,NaN,300.0,35.0,7.0,12.0
...,...,...,...,...,...,...,...,...,...
32515,b0a0c5dd78e644373b199380612c350a,NaN,NaN,NaN,NaN,1800.0,30.0,20.0,70.0
32589,10dbe0fbaa2c505123c17fdc34a63c56,NaN,NaN,NaN,NaN,800.0,30.0,10.0,23.0
32616,bd2ada37b58ae94cc838b9c0569fecd8,NaN,NaN,NaN,NaN,200.0,21.0,8.0,16.0
32772,fa51e914046aab32764c41356b9d4ea4,NaN,NaN,NaN,NaN,1300.0,45.0,16.0,45.0


In [53]:
products.duplicated().sum()

np.int64(0)

In [54]:
products["product_category_name"].value_counts().head(10)

product_category_name
cama_mesa_banho           3029
esporte_lazer             2867
moveis_decoracao          2657
beleza_saude              2444
utilidades_domesticas     2335
automotivo                1900
informatica_acessorios    1639
brinquedos                1411
relogios_presentes        1329
telefonia                 1134
Name: count, dtype: int64

In [68]:
products["product_category_name"].nunique()

73

### Первичные наблюдения

- Каждая строка таблицы соответствует одному товару.
- Столбец `product_id` является уникальным идентификатором товара.
- Полных дубликатов строк не обнаружено.
- В таблице присутствуют пропуски.
- Наибольшее количество пропусков наблюдается в столбцах, описывающих категорию товара и его характеристики (`product_category_name`, `product_name_length`, `product_description_length`, `product_photos_qty`) — по 610 записей.
- В столбцах с физическими характеристиками товара (`product_weight_g`, `product_length_cm`, `product_height_cm`, `product_width_cm`) обнаружено всего по 2 пропуска.
- Проверка показала, что все 610 пропусков в столбце `product_category_name` сопровождаются отсутствием данных в столбцах `product_name_lenght`, `product_description_lenght` и `product_photos_qty`.
- При этом физические характеристики товаров (вес и размеры) в большинстве случаев заполнены.
- Это свидетельствует о систематическом характере пропусков, а не о случайной потере отдельных значений.

# Sellers Dataset

In [56]:
sellers.shape

(3095, 4)

In [57]:
sellers.info()

<class 'pandas.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype
---  ------                  --------------  -----
 0   seller_id               3095 non-null   str  
 1   seller_zip_code_prefix  3095 non-null   int64
 2   seller_city             3095 non-null   str  
 3   seller_state            3095 non-null   str  
dtypes: int64(1), str(3)
memory usage: 96.8 KB


In [58]:
sellers["seller_id"].is_unique

True

In [59]:
sellers.duplicated().sum()

np.int64(0)

In [60]:
sellers.isna().sum()

seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

In [62]:
sellers["seller_state"].value_counts().head(10)

seller_state
SP    1849
PR     349
MG     244
SC     190
RJ     171
RS     129
GO      40
DF      30
ES      23
BA      19
Name: count, dtype: int64

In [63]:
sellers["seller_city"].value_counts().head(10)

seller_city
sao paulo         694
curitiba          127
rio de janeiro     96
belo horizonte     68
ribeirao preto     52
guarulhos          50
ibitinga           49
santo andre        45
campinas           41
maringa            40
Name: count, dtype: int64

### Первичные наблюдения

- Каждая строка таблицы соответствует одному продавцу.
- Столбец `seller_id` является уникальным идентификатором продавца.
- Полных дубликатов строк не обнаружено.
- Пропусков в таблице не обнаружено.
- Более половины продавцов зарегистрированы в штате `SP` (1849 из 3095), что указывает на высокую концентрацию продавцов в данном регионе.
- Наибольшее количество продавцов зарегистрировано в городе `sao paulo` (694 продавца).

# Product_category_name_translation Dataset

In [64]:
product_category_name_translation.shape

(71, 2)

In [65]:
product_category_name_translation.info()

<class 'pandas.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype
---  ------                         --------------  -----
 0   product_category_name          71 non-null     str  
 1   product_category_name_english  71 non-null     str  
dtypes: str(2)
memory usage: 1.2 KB


In [70]:
product_category_name_translation.duplicated().sum()

np.int64(0)

In [71]:
product_category_name_translation["product_category_name"].is_unique

True

In [72]:
product_category_name_translation["product_category_name_english"].is_unique

True

In [69]:
set(products["product_category_name"].dropna()) - set(product_category_name_translation["product_category_name"])

{'pc_gamer', 'portateis_cozinha_e_preparadores_de_alimentos'}

### Первичные наблюдения

- Таблица содержит соответствие между названиями категорий товаров на португальском и английском языках.
- Каждая категория представлена единственной строкой.
- Пропуски и полные дубликаты отсутствуют.
- Названия категорий на обоих языках являются уникальными.
- При сравнении с таблицей `products` обнаружено, что для двух категорий (`pc_gamer` и `portateis_cozinha_e_preparadores_de_alimentos`) отсутствует английский перевод. Это необходимо учитывать при объединении таблиц, так как после `merge` для этих категорий будет получено значение `NaN`.

# Итоги первичного обзора данных

В ходе первичного анализа были изучены все таблицы датасета Olist.

Проверены:
- размеры таблиц;
- типы данных;
- наличие пропусков;
- наличие полных дубликатов;
- уникальность первичных и составных ключей;
- согласованность отдельных таблиц между собой.

Основные наблюдения:

- В нескольких таблицах присутствуют пропуски, обусловленные особенностями бизнес-процесса (например, недоставленные заказы или товары без категории).
- Все столбцы с датами имеют строковый тип (`object`) и потребуют преобразования в `datetime`.
- Для двух категорий товаров отсутствует английский перевод в таблице `product_category_name_translation`.
- Существенных проблем, препятствующих дальнейшему анализу, на этапе первичного обзора не обнаружено.